src/CookBlog.Infrastructure/DAL/Services/BlobStorageService.cs

```diff
@@ -0,0 +1,71 @@
+﻿using Azure.Storage.Blobs.Models;
+using Azure.Storage.Blobs;
+using CookBlog.Api.Core.Exceptions;
+using CookBlog.Api.Core.ValuesObjects;
+using CookBlog.Core.Abstractions;
+using CookBlog.Core.DTO;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Options;
+using CookBlog.Core.Exceptions;
+
+namespace CookBlog.Infrastructure.DAL.Services;
+
+public class BlobStorageService : IBlobStorageService
+{
+    private readonly AzureOptions _azureOptions;
+    private readonly ExtensionFileOptions _extensionFileOptions;
+    private readonly BlobServiceClient _blobServiceClient;
+
+    public BlobStorageService(IOptions<AzureOptions> azureOptions,
+        ExtensionFileOptions extensionFileOptions)
+    {
+        _azureOptions = azureOptions.Value;
+        _extensionFileOptions = extensionFileOptions;
+        _blobServiceClient = new BlobServiceClient(_azureOptions.ConnectionString);
+    }
+
+    public async Task<FileDto> GetBlobImageAsync(ImagePath image)
+    {
+        if (image == null)
+        {
+            throw new InvalidImagePathException(image);
+        }
+
+        var containerClient = _blobServiceClient.GetBlobContainerClient(_azureOptions.Container);
```

Comment:
https://learn.microsoft.com/en-us/dotnet/azure/sdk/dependency-injection?tabs=web-app-builder masz jak zarejestrować to w DI